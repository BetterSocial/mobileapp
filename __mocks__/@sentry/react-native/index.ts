export const addBreadcrumb = jest.fn();
