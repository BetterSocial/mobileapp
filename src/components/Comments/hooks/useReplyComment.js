

const useReplyComment = () => {

    const isLast = (item, index, countComment) => (
      index === countComment - 1 && (item.children_counts.comment || 0) === 0
    );

    const isLastInParent = (index, countComment) => index === countComment - 1;

  return {
    isLast,
    isLastInParent
  }

}

export default useReplyComment