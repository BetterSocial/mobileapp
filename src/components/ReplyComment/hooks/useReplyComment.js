import React from 'react';
import Toast from 'react-native-simple-toast';
import moment from 'moment';
import {useNavigation} from '@react-navigation/core';
import StringConstant from '../../../utils/string/StringConstant';
import useUpdateComment from '../../Comments/hooks/useUpdateComment';
import {Context} from '../../../context';
import {createChildComment} from '../../../service/comment';
import {getFeedDetail} from '../../../service/post';
import {getCommentChild} from '../../../service/feeds';

const useReplyComment = ({
  itemProp,
  indexFeed,
  dataFeed,
  updateParent,
  updateReply,
  itemParent,
  page
}) => {
  const [temporaryText, setTemporaryText] = React.useState('');
  const [textComment, setTextComment] = React.useState('');
  const [newCommentList, setNewCommentList] = React.useState([]);
  const [item, setItem] = React.useState(itemProp);
  const navigation = useNavigation();
  // const dispatch = useDispatch
  const scrollViewRef = React.useRef(null);
  const {updateComment} = useUpdateComment();
  const [profile] = React.useContext(Context).profile;
  const [defaultData] = React.useState({
    data: {count_downvote: 0, count_upvote: 0, text: textComment},
    id: newCommentList?.length + 1,
    kind: 'comment',
    updated_at: moment(),
    children_counts: {comment: 0},
    latest_children: {},
    user: {
      data: {
        ...itemProp.user.data,
        profile_pic_url: profile?.myProfile?.profile_pic_path,
        username: profile.myProfile.username
      },
      id: itemProp.user.id
    }
  });

  const initTextComment = (text) => {
    setTextComment(text);
  };
  const setCommentHook = (text) => {
    setTemporaryText(text);
  };

  const handleFirstTextCommentHook = () => {
    setTextComment(temporaryText);
  };

  const updateReplyPost = (comment, itemParentProps, commentId) => {
    if (itemParentProps) {
      const updateMyComment = itemParentProps.latest_children.comment.map((dComment) => {
        if (dComment.id === commentId) {
          return {
            ...dComment,
            latest_children: {...dComment.latest_children, comment},
            children_counts: {comment: comment.length}
          };
        }
        return {...dComment};
      });
      const replaceComment = {
        ...itemParentProps,
        latest_children: {...itemParentProps.latest_children, comment: updateMyComment}
      };
      setItem(replaceComment);
      setNewCommentList(updateMyComment);
    }
  };

  const getThisCommentHook = () => {
    let comments = [];
    if (
      itemProp.latest_children &&
      itemProp.latest_children.comment &&
      Array.isArray(itemProp.latest_children.comment)
    ) {
      comments = itemProp.latest_children.comment.sort(
        (a, b) => moment(a.updated_at).unix() - moment(b.updated_at).unix()
      );
    }
    return comments;
  };

  const updateReplyPostHook = (comment, itemParentProps, commentId) => {
    if (itemParentProps) {
      const updateMyComment = itemParentProps.latest_children.comment.map((dComment) => {
        if (dComment.id === commentId) {
          return {
            ...dComment,
            latest_children: {...dComment.latest_children, comment},
            children_counts: {comment: comment.length}
          };
        }
        return {...dComment};
      });
      const replaceComment = {
        ...itemParentProps,
        latest_children: {...itemParentProps.latest_children, comment: updateMyComment}
      };
      return {replaceComment, updateMyComment};
    }
    return {replaceComment: itemParentProps, updateMyComment: itemParentProps};
  };

  const isLastInParentHook = (index) => index === (item.children_counts.comment || 0) - 1;

  const findCommentAndUpdateHook = (id, data) => {
    const newComment = newCommentList.map((comment) => {
      if (comment.id === id) {
        return {...comment, data: data.data};
      }
      return {...comment};
    });
    setNewCommentList(newComment);
    return newComment;
  };

  const updateVoteParentPostHook = (data, dataVote, comment) => {
    const updateMyComment = comment.latest_children.comment.map((dComment) => {
      if (dComment.id === dataVote.activity_id) {
        return {...dComment, data: data.data.data};
      }
      return {...dComment};
    });
    setNewCommentList(updateMyComment);
    return updateMyComment;
  };

  const updateVoteLatestChildrenParentHook = (response, dataVote, comment) => {
    if (comment) {
      const updateData = comment.latest_children.comment.map((dComment) => {
        if (dComment.id === dataVote.parent) {
          const mapChildren = dComment.latest_children.comment.map((child) => {
            if (child.id === dataVote.id) {
              return {...child, data: response.data};
            }
            return {...child};
          });
          return {
            ...dComment,
            latest_children: {...dComment.latest_children, comment: mapChildren}
          };
        }
        return {...dComment};
      });
      setNewCommentList(updateData);
      return updateData;
    }
    return [];
  };

  const showChildrenCommentView = async (itemReply) => {
    const itemParentProps = await {
      ...itemProp,
      latest_children: {...itemProp.latest_children, comment: newCommentList}
    };
    navigation.push('ReplyComment', {
      item: itemReply,
      level: 2,
      indexFeed,
      dataFeed,
      updateParent,
      itemParent: itemParentProps,
      updateReply: (comment, parentProps, id) => updateReplyPost(comment, parentProps, id),
      updateVote: (data, dataVote) => updateVoteParentPostHook(data, dataVote, itemParentProps),
      updateVoteLatestChildren: (data, dataVote) =>
        updateVoteLatestChildrenParentHook(data, dataVote, itemParentProps),
      getComment: getThisComment
    });
  };

  const updateFeed = async (isSort) => {
    try {
      const data = await getFeedDetail(item.activity_id);

      handleUpdateFeed(data, isSort);
    } catch (e) {
      if (__DEV__) {
        console.log(e);
      }
    }
  };
  const handleUpdateFeed = (data, isSort) => {
    if (data) {
      let oldData = data.data;
      if (isSort) {
        oldData = {
          ...oldData,
          latest_reactions: {...oldData.latest_reactions, comment: oldData.latest_reactions.comment}
        };
      }

      if (updateParent) {
        updateParent(oldData);
      }
    }
  };
  const createComment = async (isAnonimity, anonimityData) => {
    let sendPostNotif = false;
    if (page !== 'DetailDomainScreen') {
      sendPostNotif = true;
    }

    const commentWillBeAddedData = {
      ...defaultData,
      data: {...defaultData.data, text: textComment}
    };
    if (isAnonimity) {
      commentWillBeAddedData.user = {};
      commentWillBeAddedData.data.is_anonymous = true;
      commentWillBeAddedData.data.anon_user_info_emoji_name = anonimityData.emojiName;
      commentWillBeAddedData.data.anon_user_info_emoji_code = anonimityData.emojiCode;
      commentWillBeAddedData.data.anon_user_info_color_name = anonimityData.colorName;
      commentWillBeAddedData.data.anon_user_info_color_code = anonimityData.colorCode;
    }
    if (!isAnonimity) {
      commentWillBeAddedData.user.data = {
        profile_pic_url: profile?.myProfile?.profile_pic_path,
        username: profile?.myProfile?.username
      };
    }

    setTemporaryText('');
    setNewCommentList([...newCommentList, commentWillBeAddedData]);

    try {
      if (textComment.trim() !== '') {
        const data = await createChildComment(
          textComment,
          item.id,
          item.user.id,
          sendPostNotif,
          dataFeed?.actor?.id,
          dataFeed.id,
          dataFeed.message,
          isAnonimity,
          anonimityData
        );
        scrollViewRef.current.scrollToEnd();
        if (data.code === 200) {
          const newComment = [
            ...newCommentList,
            {
              ...defaultData,
              id: data.data.id,
              activity_id: data.data.activity_id,
              user: data.data.user,
              data: data.data.data,
              is_you: true
            }
          ];
          setNewCommentList(newComment);
          if (typeof updateReply === 'function') {
            updateReply(newComment, itemParent, item.id);
          }
          updateFeed(true);
          updateComment(item.activity_id);
        } else {
          Toast.show(StringConstant.generalCommentFailed, Toast.LONG);
        }
      } else {
        // Toast.show('Comments are not empty', Toast.LONG);
        // setLoadingCMD(false);
      }
    } catch (error) {
      Toast.show(StringConstant.generalCommentFailed, Toast.LONG);
    }
  };
  const getThisComment = async (isUpdate) => {
    if (itemProp.latest_children.comment && Array.isArray(itemProp.latest_children.comment)) {
      if (!isUpdate) {
        setNewCommentList(itemProp.latest_children?.comment);
      }
    }
    const response = await getCommentChild({activity_id: item?.id, feed_id: item.activity_id});
    setNewCommentList(response.data);
  };

  return {
    getThisCommentHook,
    updateReplyPostHook,
    setTemporaryText,
    setCommentHook,
    temporaryText,
    handleFirstTextCommentHook,
    textComment,
    isLastInParentHook,
    findCommentAndUpdateHook,
    updateVoteParentPostHook,
    updateVoteLatestChildrenParentHook,
    setTextComment,
    newCommentList,
    setNewCommentList,
    defaultData,
    setItem,
    item,
    initTextComment,
    updateReplyPost,
    showChildrenCommentView,
    updateFeed,
    handleUpdateFeed,
    scrollViewRef,
    createComment,
    getThisComment
  };
};
export default useReplyComment;
