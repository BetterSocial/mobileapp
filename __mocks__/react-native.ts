import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN