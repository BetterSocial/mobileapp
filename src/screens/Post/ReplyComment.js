import * as React from 'react';
import {Alert, StyleSheet, Text, TouchableOpacity, View} from 'react-native';

import Toast from 'react-native-simple-toast';
import {useNavigation} from '@react-navigation/native';

import Comment from '../../elements/PostDetail/Comment';
import WriteComment from '../../elements/PostDetail/WriteComment';
import ArrowLeftIcon from '../../../assets/icons/arrow-left.svg';
import {fonts} from '../../utils/fonts';
import {colors} from '../../utils/colors';
import {createChildComment} from '../../service/comment';

const ReplyComment = (props) => {
  const navigation = useNavigation();
  const [item, setItem] = React.useState(props.route.params.item);
  const [parent, setParent] = React.useState(props.route.params.parent);
  const [textComment, setTextComment] = React.useState('');
  const [isReaction, setReaction] = React.useState(false);

  React.useEffect(() => {
    const init = () => {
      if (JSON.stringify(item.children_counts) !== {}) {
        setReaction(true);
      }
    };
    init();
  }, [item]);

  const createComment = async () => {
    if (textComment === '') {
      Alert.alert('warning', 'Comment not null');
      return;
    }

    try {
      let data = await createChildComment(textComment, item.id);
      console.log(data);
      if (data.code === 200) {
        setTextComment('');
        Toast.show('Successfully Comment', Toast.LONG);
        navigation.goBack();
      } else {
        Toast.show('Failed Comment', Toast.LONG);
      }
    } catch (error) {
      Toast.show('Failed Comment', Toast.LONG);
      console.log(error);
    }
  };
  return (
    <View style={styles.container}>
      {/* <ScrollView> */}
      <View style={styles.header}>
        <TouchableOpacity>
          <ArrowLeftIcon width={20} height={12} fill="#000" />
        </TouchableOpacity>
        <Text style={styles.headerText}>
          Reply to{' '}
          {/* {item.anonimiti === true ? Anonymous : item.actor.data.username} */}
          {item.user.data.username}
        </Text>
        <View style={styles.btn} />
      </View>
      <View style={styles.containerComment}>
        <Comment
          username={item.user.data.username}
          comment={item.data.text}
          onPress={() => {}}
        />
        {item.children_counts.comment > 0 &&
          item.latest_children.comment.map((itemReply, index) => {
            return (
              <ContainerReply>
                <Comment
                  key={'r' + index}
                  username={itemReply.user.data.username}
                  comment={itemReply.data.text}
                  onPress={() => {}}
                />
                {itemReply.children_counts.comment > 0 &&
                  itemReply.latest_children.comment.map(
                    (itemReplyChild, index) => {
                      return (
                        <ContainerReply>
                          <Comment
                            key={'rc' + index}
                            username={itemReplyChild.user.data.username}
                            comment={itemReplyChild.data.text}
                            onPress={() => {}}
                          />
                          {}
                        </ContainerReply>
                      );
                    },
                  )}
              </ContainerReply>
            );
          })}
      </View>
      {/* <Text style={styles.post}>text commentParent</Text> */}
      {/* {isReaction && <ContainerComment comments={item.latest_children} />} */}
      {/* <View style={styles.comment}>
        <Image
          source={require('../.../../../assets/images/ProfileDefault.png')}
          style={styles.image}
        />
        <TextInput
          style={styles.input}
          placeholder="Your comment"
          multiline={true}
          numberOfLines={4}
          textAlignVertical="top"
          onChangeText={(v) => setTextComment(v)}
          value={textComment}
        />
      </View> */}
      <WriteComment
        onChangeText={(v) => setTextComment(v)}
        onPress={() => createComment()}
        value={textComment}
      />
      {/* </ScrollView> */}
    </View>
  );
};
const ContainerReply = ({children, isGrandchild}) => {
  return (
    <View
      style={[
        styles.containerReply,
        {borderColor: isGrandchild ? '#fff' : colors.gray1},
      ]}>
      {children}
    </View>
  );
};
export default ReplyComment;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  containerComment: {paddingHorizontal: 20, marginTop: 24.5},
  header: {
    paddingTop: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 22,
    alignItems: 'center',
  },
  containerReply: {
    borderLeftWidth: 1,
    paddingLeft: 30,
  },
  btn: {
    paddingVertical: 8,
    paddingHorizontal: 20,
    borderRadius: 20,
  },
  btnText: {
    color: '#fff',
  },
  headerText: {
    fontFamily: fonts.inter[600],
    fontSize: 14,
    color: '#000',
  },
  image: {
    width: 48,
    height: 48,
  },
  input: {
    backgroundColor: '#F2F2F2',
    flex: 1,
    color: '#000',
    padding: 10,
    marginLeft: 20,
    borderRadius: 8,
  },
  comment: {
    flexDirection: 'row',
    paddingRight: 20,
    position: 'absolute',
    bottom: 0,
  },
  post: {
    fontFamily: fonts.inter[400],
    fontSize: 16,
    color: '#333333',
    marginLeft: 28,
  },
});
