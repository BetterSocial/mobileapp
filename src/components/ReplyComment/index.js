import * as React from 'react';
import {
  Pressable,
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import {useNavigation} from '@react-navigation/native';

import ArrowLeftIcon from '../../../assets/icons/arrow-left.svg';
import ButtonHightlight from '../ButtonHighlight';
import Comment from '../Comments/Comment';
import ConnectorWrapper from '../Comments/ConnectorWrapper';
import ReplyCommentItem from '../Comments/ReplyCommentItem';
import StringConstant from '../../utils/string/StringConstant';
import WriteComment from '../Comments/WriteComment';
import useReplyComment from './hooks/useReplyComment';
import useWriteComment from '../Comments/hooks/useWriteComment';
import {fonts} from '../../utils/fonts';
import useCommentAction from '../Comments/hooks/useCommentAction';
import {COLORS} from '../../utils/theme';

const ReplyCommentId = ({
  itemProp,
  indexFeed,
  level,
  updateParent,
  page,
  dataFeed,
  updateReply,
  itemParent,
  getComment
}) => {
  const navigation = useNavigation();
  const {
    setCommentHook,
    temporaryText,
    isLastInParentHook,
    setTextComment,
    newCommentList,
    item,
    showChildrenCommentView,
    updateFeed,
    scrollViewRef,
    createComment,
    getThisComment
  } = useReplyComment({
    itemProp,
    indexFeed,
    dataFeed,
    updateParent,
    updateReply,
    itemParent,
    page,
    getComment
  });
  const {handleUsernameReplyComment} = useWriteComment();
  const {showAlertDelete} = useCommentAction();
  React.useEffect(() => {
    if (setTextComment && typeof setTextComment === 'function') {
      setTextComment(temporaryText);
    }
  }, [temporaryText]);

  React.useEffect(() => {
    if (itemProp) {
      getThisComment();
    }
  }, [itemProp]);

  const updateVote = () => {
    if (getComment && typeof getComment === 'function') {
      getComment();
    }
  };

  const navigationGoBack = () => navigation.goBack();
  if (!item) return null;
  return (
    <SafeAreaView style={styles.container}>
      <StatusBar translucent={false} />
      <View style={styles.header}>
        <TouchableOpacity testID="backButton" onPress={navigationGoBack} style={styles.backArrow}>
          <ArrowLeftIcon width={20} height={20} fill="#000" />
        </TouchableOpacity>
        <Text testID="usernameText" style={styles.headerText}>
          Reply to {handleUsernameReplyComment(itemProp)}
        </Text>
        <View style={styles.btn} />
      </View>
      <ScrollView ref={scrollViewRef} contentContainerStyle={styles.commentScrollView}>
        <View style={styles.containerComment}>
          <ReplyCommentItem
            feedId={dataFeed?.id}
            indexFeed={indexFeed}
            user={item.user}
            comment={item}
            time={item.created_at}
            photo={item.user?.data?.profile_pic_url}
            isLast={newCommentList.length <= 0}
            level={level}
            refreshComment={updateFeed}
            updateVoteParent={getComment}
          />
          {newCommentList.length > 0 &&
            newCommentList.map((itemReply, index) => (
              <React.Fragment key={index}>
                {itemReply.user ? (
                  <ContainerReply>
                    <ConnectorWrapper index={index}>
                      <Pressable
                        onLongPress={() => {
                          showAlertDelete(itemReply, false, () => getThisComment(true));
                        }}
                        style={styles.childCommentWrapper}>
                        <Comment
                          indexFeed={indexFeed}
                          showLeftConnector={false}
                          time={itemReply.created_at}
                          photo={itemReply.user.data && itemReply.user.data.profile_pic_url}
                          isLast={level >= 2}
                          key={`r${index}`}
                          user={itemReply.user}
                          comment={itemReply}
                          onPress={() => showChildrenCommentView(itemReply)}
                          level={parseInt(level, 10) + 1}
                          updateVote={updateVote}
                          onLongPress={() => {
                            showAlertDelete(itemReply, false, () => getThisComment(true));
                          }}
                        />
                        {itemReply.children_counts.comment > 0 && (
                          <>
                            <View style={styles.seeRepliesContainer(isLastInParentHook(index))}>
                              <View style={styles.connector} />
                              <ButtonHightlight onPress={() => showChildrenCommentView(itemReply)}>
                                <Text style={styles.seeRepliesText}>
                                  {StringConstant.postDetailPageSeeReplies(
                                    itemReply.children_counts.comment || 0
                                  )}
                                </Text>
                              </ButtonHightlight>
                            </View>
                          </>
                        )}
                      </Pressable>
                    </ConnectorWrapper>
                  </ContainerReply>
                ) : null}
              </React.Fragment>
            ))}
          {newCommentList.length > 0 ? <View style={styles.childLevelMainConnector} /> : null}
        </View>
      </ScrollView>
      <WriteComment
        postId={item?.activity_id}
        inReplyCommentView={true}
        showProfileConnector={newCommentList.length > 0}
        username={handleUsernameReplyComment(itemProp)}
        onChangeText={setCommentHook}
        onPress={(isAnonimity, anonimityData) => {
          createComment(isAnonimity, anonimityData);
        }}
        value={temporaryText}
      />
    </SafeAreaView>
  );
};
export const ContainerReply = ({children, isGrandchild = true, key}) => (
  <View
    key={key}
    style={[
      styles.containerReply,
      {borderColor: isGrandchild ? COLORS.transparent : COLORS.balance_gray}
    ]}>
    {children}
  </View>
);

export default React.memo(ReplyCommentId);

export const styles = StyleSheet.create({
  container: {
    height: 'auto',
    flex: 1,
    backgroundColor: COLORS.white
    // backgroundColor: 'blue',
  },
  containerComment: {
    marginTop: 8,
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    paddingLeft: 36,
    paddingRight: 23
  },
  header: {
    marginRight: -20,
    marginBottom: 8,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  containerReply: {
    borderLeftWidth: 1,
    width: '100%'
    // flex: 1,
  },
  seeRepliesContainer: (isLast) => ({
    display: 'flex',
    flexDirection: 'row',
    paddingBottom: 14,
    borderLeftColor: isLast ? COLORS.transparent : COLORS.balance_gray
  }),
  seeRepliesText: {
    color: COLORS.signed_primary
  },
  btn: {
    paddingVertical: 8,
    paddingHorizontal: 20,
    borderRadius: 20
  },
  btnText: {
    color: COLORS.white
  },
  headerText: {
    fontFamily: fonts.inter[600],
    fontSize: 14,
    marginLeft: -24,
    color: COLORS.black,
    alignSelf: 'center'
  },
  image: {
    width: 48,
    height: 48
  },
  input: {
    backgroundColor: COLORS.concrete,
    flex: 1,
    color: COLORS.b,
    padding: 10,
    marginLeft: 20,
    borderRadius: 8
  },
  comment: {
    flexDirection: 'row',
    paddingRight: 20,
    position: 'absolute',
    bottom: 0
  },
  post: {
    fontFamily: fonts.inter[400],
    fontSize: 16,
    color: COLORS.mine_shaft,
    marginLeft: 28
  },
  mainLeftConnector: {
    height: '100%',
    width: 1,
    position: 'absolute',
    backgroundColor: COLORS.balance_gray,
    left: 46,
    zIndex: -100
  },
  connector: {
    width: 10,
    height: 5,
    borderLeftWidth: 1.5,
    borderBottomWidth: 1.5,
    // borderBottomLeftRadius: 1,
    borderLeftColor: COLORS.balance_gray,
    borderBottomColor: COLORS.balance_gray,
    marginRight: 4,
    marginLeft: -1,
    borderBottomLeftRadius: 15 / 2,
    marginTop: 0
    // marginTop: 1
  },
  childCommentWrapper: {
    borderLeftColor: COLORS.balance_gray,
    borderLeftWidth: 1,
    flex: 1
  },
  childLevelMainConnector: {
    flex: 1,
    borderLeftWidth: 1,
    borderLeftColor: COLORS.balance_gray,
    marginLeft: 24
  },
  backArrow: {
    padding: 10,
    paddingLeft: 24,
    paddingVertical: 18,
    alignSelf: 'center'
  },
  commentScrollView: {
    minHeight: '100%',
    paddingBottom: 83
  },
  childCommentWrapperLoading: {
    flex: 1
  }
});
