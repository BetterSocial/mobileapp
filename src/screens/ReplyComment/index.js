import * as React from 'react';
import {StyleSheet, View} from 'react-native';

import ReplyCommentComponent from '../../components/ReplyComment';
import {Context} from '../../context';
import {setFeedByIndex} from '../../context/actions/feeds';
import {withInteractionsManaged} from '../../components/WithInteractionManaged';

const ReplyComment = (props) => {
  // const [newComment, setNewComment] = React.useState([])
  const itemProp = props.route.params.item;
  const {
    updateParent,
    findCommentAndUpdate,
    updateReply,
    itemParent,
    parentComment,
    updateVote,
    updateVoteLatestChildren
  } = props.route.params;
  const {level} = props.route.params;
  const {dataFeed} = props.route.params;
  const [feeds, dispatch] = React.useContext(Context).feeds;
  const feedIndex = () => {
    if (feeds && Array.isArray(feeds)) {
      const findIndex = feeds.find((feed) => feed.id === itemProp.activity_id);
      return findIndex;
    }
    return 0;
  };
  return (
    <View style={styles.container}>
      <ReplyCommentComponent
        indexFeed={feedIndex()}
        itemProp={itemProp}
        dispatch={dispatch}
        feeds={feeds.feeds}
        level={level}
        setFeedByIndexProps={setFeedByIndex}
        updateParent={updateParent}
        page={props.route.params.page}
        dataFeed={dataFeed}
        updateReply={updateReply}
        updateVote={updateVote}
        updateVoteLatestChildren={updateVoteLatestChildren}
        findCommentAndUpdate={findCommentAndUpdate}
        itemParent={itemParent}
        parentComment={parentComment}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: '100%'
  }
});

export default withInteractionsManaged(ReplyComment);
