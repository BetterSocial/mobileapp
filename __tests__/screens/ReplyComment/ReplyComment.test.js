import React from 'react'
import {render} from '@testing-library/react-native'
import {act, renderHook} from '@testing-library/react-hooks'
import ReplyComment from '../../../src/screens/ReplyComment'
import useReplyComment from '../../../src/components/ReplyComment/hooks/useReplyComment'

const itemProp = {
    latest_children: {
        comment: [
            {
                activity_id: "bea7567e-38f1-11ed-8bdd-0e4b8d0e7a11",
                updated_at: "2022-09-22T00:20:23.011138Z"

            },
            {
                activity_id: "bea7567e-38f1-11ed-8bdd-0e4b8d0e7a11",
                updated_at: "2022-09-21T00:20:23.011138Z"

            }
        ]
    },
    children_counts: {
        comment: 2
    }
}

describe('it should same as snapshot', () => {
    it('code should not change', () => {
        const spyIntreaction = jest.spyOn(React, 'useRef').mockReturnValueOnce({current: {}})
        const container = render(<ReplyComment />)
        expect(container).toMatchSnapshot()
        expect(spyIntreaction).toHaveBeenCalled()
    })
})

describe('hooks function should run correctly', () => {
    it('should sort date correctly', () => {
        const {result} = renderHook(() => useReplyComment())
        expect(result.current.getThisCommentHook(itemProp)).toStrictEqual([
            {
                activity_id: "bea7567e-38f1-11ed-8bdd-0e4b8d0e7a11",
                updated_at: "2022-09-21T00:20:23.011138Z"

            },
            {
                  activity_id: "bea7567e-38f1-11ed-8bdd-0e4b8d0e7a11",
                updated_at: "2022-09-22T00:20:23.011138Z"

            }
        ])
    })


    it('handle temporary comment should run correctly', () => {
        const {result} = renderHook(() => useReplyComment())
        act(() => {
            result.current.setCommentHook('test')
        })
        expect(result.current.temporaryText).toStrictEqual('test')

    })

    it('isLastInParent should run correctly', () => {
        const {result} = renderHook(() => useReplyComment())
        expect(result.current.isLastInParentHook(1, itemProp)).toStrictEqual(true)
        expect(result.current.isLastInParentHook(0, itemProp)).toStrictEqual(false)
    })

    // it('should update parent reply should run correctly', () => {
    //     const {result} = renderHook(() => useReplyComment())
    //     expect(result.current.updateReplyPostHook("test", itemProp, "bea7567e-38f1-11ed-8bdd-0e4b8d0e7a11").updateComment).toStrictEqual({
    //         latest_children: {
    //         comment: [
    //             {
    //                 activity_id: "bea7567e-38f1-11ed-8bdd-0e4b8d0e7a12",
    //                 updated_at: "2022-09-22T00:20:23.011138Z"

    //             },
    //             {
    //                 activity_id: "bea7567e-38f1-11ed-8bdd-0e4b8d0e7a11",
    //                 updated_at: "2022-09-21T00:20:23.011138Z",
    //                 comment: "test"

    //             }
    //         ]
    // },

    //     })
    // })
})