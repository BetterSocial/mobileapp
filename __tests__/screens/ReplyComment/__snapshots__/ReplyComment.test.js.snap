// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should same as snapshot code should not change 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  transition={
    <Transition.Together>
      <Transition.Change
        interpolation="easeInOut"
      />
      <Transition.In
        type="fade"
      />
    </Transition.Together>
  }
/>
`;
